optuna/study.py

```diff
@@ -472,14 +473,34 @@ def create_study(
         direction:
             Direction of optimization. Set ``minimize`` for minimization and ``maximize`` for
             maximization. Note that ``maximize`` is currently unsupported.
+        exist_ok:
+            Flag for controlling the behavior when study names conflict.
+            If a study named ``study_name`` already exists in the ``storage``,
+            a :class:`~optuna.structs.DuplicatedStudyError` is raised if ``exist_ok`` is
+            set to :obj:`False`.
+            Otherwise the creation of the study is skipped and no exception is raised.
 
     Returns:
         A :class:`~optuna.study.Study` object.
 
     """
 
     storage = storages.get_storage(storage)
-    study_name = storage.get_study_name_from_id(storage.create_new_study_id(study_name))
+    try:
+        study_id = storage.create_new_study_id(study_name)
+    except structs.DuplicatedStudyError:
+        if exist_ok:
+            assert study_name is not None
+
+            logger = logging.get_logger(__name__)
+            logger.info('The same name study exists in the storage. '
```

Comment:
`storage` should be hidden from users if possible. I think we can rephrase this sentence as follows:

`'The study name {} is duplicated with the existing study.'.format(study_name)`